server/lib/payments.js

```diff
@@ -391,11 +391,27 @@ export async function associateTransactionRefundId(transaction, refund, data) {
   return find([refundCredit, refundDebit, debit, credit], { id: transaction.id });
 }
 
+/*
+ * Send email notifications.
+ *
+ * Don't send emails when moving funds between a sub-collective(event/project) and its parent or
+ * from between a host and one of its collectives. Transaction types for each of these are:
+ *
+ * Child to Parent Collective: CREDIT
+ * Host to its own collective: CREDIT
+ * Parent Collective to Child: CREDIT
+ * Collective to its Host: CREDIT
+ *
+ */
 export const sendEmailNotifications = (order, transaction) => {
   debug('sendEmailNotifications');
-  // for gift cards and manual payment methods
-  // Don't send a "thank you" email when moving funds between a sub-collective and its parent
-  if (transaction && order.fromCollective?.ParentCollectiveId !== order.collective?.id) {
+  if (
+    transaction &&
+    order.fromCollective?.ParentCollectiveId !== order.collective?.id &&
```

Comment:
I was expecting to see some comments before each case. It's not really readable having the comments on top. 

I was also expecting detailed information about what is `fromCollective` and `collective` in each case.

We're talking for days about making this code block understandable and we're still not there. Can you make sure it's understandable by someone not familiar with the code base?

Reply:
@znarf : Ops, sorry for the trouble. Let me try to make it better. I've added comments at each case explaining `fromCollective` and `collective` in each case. Let me know if you think if there's anything to expand on in the comments.